src/Features/Core/Portable/DesignerAttributes/AbstractDesignerAttributeService.cs

```diff
@@ -67,6 +67,11 @@ protected AbstractDesignerAttributeService(Workspace workspace)
         private async Task<DesignerAttributeResult> ScanDesignerAttributesInRemoteHostAsync(RemoteHostClient client, Document document, CancellationToken cancellationToken)
         {
             var keepAliveSession = await TryGetKeepAliveSessionAsync(client, cancellationToken).ConfigureAwait(false);
+            if (keepAliveSession == null)
+            {
+                // The client is not currently running, so we don't know the state of the DesignerAttribute.
+                return new DesignerAttributeResult(designerAttributeArgument: null, containsErrors: false, notApplicable: true);
```

Comment:
Is this the same error result as if TryInvokeAsync below also fails? Can we make this pattern a bit less error prone?

Reply:
we might be able to provide 1 API that does all checking for caller, but someone has to check each step due to user can kill OOP process at any moment.

caller usually need to do this 3 steps.

1. get client (which represent 1 service hub process)
2. get a session (which represent 1 service the the process)
3. TryInvoke (which represent running 1 method call in the service in the process)

each step can could fail but not likely unless things like OOM happens except when users kills OOP directly.

in that case, things can fail between each steps. so we need to guard each step. but sure we can hide it from caller.